src/main/java/com/mongodb/kafka/connect/util/ConnectionValidator.java

```diff
@@ -0,0 +1,174 @@
+            MongoClientSettings mongoClientSettings = MongoClientSettings.builder()
+                    .applyConnectionString(new ConnectionString((String) configValue.value()))
+                    .applyToClusterSettings(b -> b.addClusterListener(new ClusterListener() {
+                        @Override
+                        public void clusterOpening(final ClusterOpeningEvent event) {
+                        }
+
+                        @Override
+                        public void clusterClosed(final ClusterClosedEvent event) {
+                        }
+
+                        @Override
+                        public void clusterDescriptionChanged(final ClusterDescriptionChangedEvent event) {
+                            if (!connected.get() && event.getNewDescription().hasReadableServer(ReadPreference.primaryPreferred())) {
+                                connected.set(true);
+                                latch.countDown();
+
+                            }
+                        }
+                    }))
+                    .build();
+
+            long latchTimeout = mongoClientSettings.getSocketSettings().getConnectTimeout(TimeUnit.MILLISECONDS) + 500;
+            MongoClient mongoClient = MongoClients.create(mongoClientSettings);
+
+            try {
+                if (!latch.await(latchTimeout, TimeUnit.MILLISECONDS)) {
+                    throw new Exception("Didn't connect");
+                }
+            } catch (Exception e) {
+                configValue.addErrorMessage("Invalid connection string, unable to reach the server.");
+                mongoClient.close();
+            }
+            if (configValue.errorMessages().isEmpty()) {
+                return Optional.of(mongoClient);
+            }
+        }
+        return Optional.empty();
+    }
+
+    public static void validateHasChangeStreamPermissions(final MongoClient mongoClient, final MongoSourceConfig sourceConfig,
+                                                          final Config config) {
+
+        String database = sourceConfig.getString(MongoSourceConfig.DATABASE_CONFIG);
+        String collection = sourceConfig.getString(MongoSourceConfig.COLLECTION_CONFIG);
+
+        Optional<List<Document>> pipeline = sourceConfig.getPipeline();
+        ChangeStreamIterable<Document> changeStream;
+        if (database.isEmpty()) {
+            changeStream = pipeline.map(mongoClient::watch).orElse(mongoClient.watch());
+        } else if (collection.isEmpty()) {
+            MongoDatabase db = mongoClient.getDatabase(database);
+            changeStream = pipeline.map(db::watch).orElse(db.watch());
+        } else {
+            MongoCollection<Document> coll = mongoClient.getDatabase(database).getCollection(collection);
+            changeStream = pipeline.map(coll::watch).orElse(coll.watch());
+        }
+
+        try (MongoChangeStreamCursor<ChangeStreamDocument<Document>> cursor = changeStream.cursor()) {
+            cursor.tryNext();
```

Comment:
For changestreams we can just try to create one and see if the user has permissions.